### PR TITLE
Use wget to download rpi-update

### PR DESCRIPTION
**Pull request by xipmix**
I have had quite a lot of certificate validation errors with curl pulling down rpi-update, it was failing with
```
curl: (60) SSL certificate problem: unable to get local issuer certificate
```
I checked the certificates and ca-certificates.crt in the rootfs that was created by raspberryconfig.sh. Everything required was there and correct, I manually verified the certificate chain and root CA. I could not track down the issue that causes curl to fail. Switching to wget fixed the problem. Since this is the only place curl is actually used, this would seem to be the right answer.

### CODE REVIEW

**Comment by volumio** [comment]
On which system are you building? It's a VM or a proper host? Do you have the time set properly on the host machine?

**Comment by xipmix** [comment]
It's a physical host with ubuntu lts 14.04 amd64, with kernel 4.4.0  
(the official ubuntu backport). Hacking volumio is all it does.      
I am building with:                                                  
```                                                                  
sudo ./build.sh -b arm -d pi -v 2.0                                  
```                                                                  

ntp setup is working ok, the host TZ is GMT+10. In the build chroot the TZ is Etc/UTC.
The epoch time matches up between the host and the build chroot.

Running the same operations (that raspberryconfig.sh does to download the file) in the host OS works every time. The curl versions are very similar (host 7.35.0, build 7.39.0), openssl differs somewhat (host 1.0.1f, build 1.0.1t).                

The strange thing is that curl sometimes works. When it fails, it fails the same way,  just after the SSL handshake:                 
```                                                                  
* Connected to raw.githubusercontent.com (151.101.100.133) port 443 (#0)
* successfully set certificate verify locations:                     
*   CAfile: none                                                     
  CApath: /etc/ssl/certs                                             
* SSLv3, TLS handshake, Client hello (1):                            
} [data not shown]                                                   
* SSLv3, TLS handshake, Server hello (2):                            
{ [data not shown]                                                   
* SSLv3, TLS handshake, CERT (11):                                   
{ [data not shown]                                                   
* SSLv3, TLS alert, Server hello (2):                                
} [data not shown]                                                   
* SSL certificate problem: unable to get local issuer certificate    
* Closing connection 0                                               
* SSLv3, TLS alert, Client hello (1):                                
} [data not shown]                                                   
```                                                                  

I bumped into this while exploring how to move the location where the volumio image is put together (/mnt/volumio) to somewhere of my choosing (I don't have enough space in the / partition).               

I'm happy for you to ignore this PR while I go back and step through my changes to figure out what is breaking things. I just thought it would be tidier to ditch the one use of curl.

**Comment by xipmix** [comment]
So I looked again and it's still something of a mystery.                        
I can work around by calling curl like this:                                    
```                                                                             
curl -L --output /usr/bin/rpi-update --cacert /etc/ssl/certs/ca-certificates.crt
```                                                                             
For some reason giving ```--capath /etc/ssl/certs``` instead does not work, I   have to use the ca bundle file.
                                                                                
What triggered all this was testing the patch below. I wanted to make it        possible to collate the image somewhere else than /mnt/volumio. It's            essentially: sed -e 's+/mnt/volumio+$TOPDIR+'.
```                                                                             
diff --git a/scripts/raspberryimage.sh b/scripts/raspberryimage.sh              
index 2bce6b0..1867cf6 100755                                                   
--- a/scripts/raspberryimage.sh                                                 
+++ b/scripts/raspberryimage.sh                                                 
@@ -12,6 +12,8 @@ while getopts ":v:p:" opt; do                                 
   esac                                                                         
 done                                                                           
                                                                                
+TOPDIR=/mnt/volumio                                                            
+echo "Volumio Temp Directory is '$TOPDIR'"                                     
 BUILDDATE=$(date -I)                                                           
 IMG_FILE="Volumio${VERSION}-${BUILDDATE}-pi.img"                               
                                                                                
@@ -54,53 +56,53 @@ else                                                        
        sudo mkdir /mnt                                                         
 fi                                                                             
                                                                                
-if [ -d /mnt/volumio ]; then                                                   
+if [ -d $TOPDIR ]; then                                                        
        echo "Volumio Temp Directory Exists - Cleaning it"                      
-       rm -rf /mnt/volumio/*                                                   
+       rm -rf $TOPDIR/*                                                        
 else                                                                           
        echo "Creating Volumio Temp Directory"                                  
-       sudo mkdir /mnt/volumio                                                 
+       sudo mkdir -p $TOPDIR                                                   
 fi                                                                             

 #Create mount point for the images partition                                   
-sudo mkdir /mnt/volumio/images                                                 
-sudo mount -t ext4 "${IMG_PART}" /mnt/volumio/images                           
-sudo mkdir /mnt/volumio/rootfs                                                 
-sudo cp -pdR build/arm/root/* /mnt/volumio/rootfs                              
-#sudo mkdir /mnt/volumio/boot                                                  
-sudo mount -t vfat "${BOOT_PART}" /mnt/volumio/rootfs/boot                     
+sudo mkdir $TOPDIR/images                                                      
+sudo mount -t ext4 "${IMG_PART}" $TOPDIR/images                                
+sudo mkdir $TOPDIR/rootfs                                                      
+sudo cp -pdR build/arm/root/* $TOPDIR/rootfs                                   
+#sudo mkdir $TOPDIR/boot                                                       
+sudo mount -t vfat "${BOOT_PART}" $TOPDIR/rootfs/boot                          
                                                                                
 sync                                                                           
                                                                                
 echo "Entering Chroot Environment"                                             
                                                                                
-cp scripts/raspberryconfig.sh /mnt/volumio/rootfs                              
+cp scripts/raspberryconfig.sh $TOPDIR/rootfs                                   

-cp scripts/initramfs/init /mnt/volumio/rootfs/root                             
-cp scripts/initramfs/mkinitramfs-custom.sh /mnt/volumio/rootfs/usr/local/sbin  
+cp scripts/initramfs/init $TOPDIR/rootfs/root                                  
+cp scripts/initramfs/mkinitramfs-custom.sh $TOPDIR/rootfs/usr/local/sbin       
                                                                                
 #copy the scripts for updating from usb                                        
-wget -P /mnt/volumio/rootfs/root http://repo.volumio.org/Volumio2/Binaries/    volumio-init-updater
+wget -P $TOPDIR/rootfs/root http://repo.volumio.org/Volumio2/Binaries/volumio- init-updater
                                                                                
-mount /dev /mnt/volumio/rootfs/dev -o bind                                     
-mount /proc /mnt/volumio/rootfs/proc -t proc                                   
-mount /sys /mnt/volumio/rootfs/sys -t sysfs                                    
+mount /dev $TOPDIR/rootfs/dev -o bind                                          
+mount /proc $TOPDIR/rootfs/proc -t proc                                        
+mount /sys $TOPDIR/rootfs/sys -t sysfs                                         
                                                                                
 echo "Custom dtoverlay pre and post"                                           
-cp -rp volumio/opt/vc/bin/* /mnt/volumio/rootfs/opt/vc/bin/                    
+cp -rp volumio/opt/vc/bin/* $TOPDIR/rootfs/opt/vc/bin/                         
                                                                                
-echo $PATCH > /mnt/volumio/rootfs/patch                                        
-chroot /mnt/volumio/rootfs /bin/bash -x <<'EOF'                                
+echo $PATCH > $TOPDIR/rootfs/patch                                             
+chroot $TOPDIR/rootfs /bin/bash -x <<'EOF'                                     
 su -                                                                           
 /raspberryconfig.sh -p                                                         
 EOF                                                                            
                                                                                
 echo "Base System Installed"                                                   
-rm /mnt/volumio/rootfs/raspberryconfig.sh /mnt/volumio/rootfs/root/init        
+rm $TOPDIR/rootfs/raspberryconfig.sh /mnt/volumio/rootfs/root/init             
 echo "Unmounting Temp devices"                                                 
-umount -l /mnt/volumio/rootfs/dev                                              
-umount -l /mnt/volumio/rootfs/proc                                             
-umount -l /mnt/volumio/rootfs/sys                                              
+umount -l $TOPDIR/rootfs/dev                                                   
+umount -l $TOPDIR/rootfs/proc                                                  
+umount -l $TOPDIR/rootfs/sys                                                   
                                                                                
                                                                                
@@ -119,7 +121,7 @@ else                                                        
 fi                                                                             
                                                                                
 echo "Copying Volumio ROOTFS to Temp DIR"                                      
-cp -rp /mnt/volumio/rootfs/* /mnt/squash/                                      
+cp -rp $TOPDIR/rootfs/* /mnt/squash/                                           
                                                                                
 echo "Removing Kernel"                                                         
 rm -rf /mnt/squash/boot/*                                                      
@@ -134,14 +136,14 @@ echo "Cleaning squash environment"                        
 rm -rf /mnt/squash                                                             
                                                                                
 #copy the squash image inside the images partition                             
-cp Volumio.sqsh /mnt/volumio/images/volumio_current.sqsh                       
+cp Volumio.sqsh $TOPDIR/images/volumio_current.sqsh                            
                                                                                
 echo "Unmounting Temp Devices"                                                 
-sudo umount -l /mnt/volumio/images                                             
-sudo umount -l /mnt/volumio/rootfs/boot                                        
+sudo umount -l $TOPDIR/images                                                  
+sudo umount -l $TOPDIR/rootfs/boot                                             
                                                                                
 echo "Cleaning build environment"                                              
-rm -rf /mnt/volumio /mnt/boot                                                  
+rm -rf $TOPDIR /mnt/boot                                                       
                                                                                
 dmsetup remove_all                                                             
 sudo losetup -d ${LOOP_DEV}                                                    
                                                                                
```                                                                             
with TOPDIR set to /mnt/volumio, curl works fine.                               
                                                                                
When I change TOPDIR to e.g. /data/me/mnt, curl falls over.                     
This happens to be on a different physical disk to the boot disk.               
                                                                                
I added an openssl check of the certificate to the script, just before the curl call
```                                                                             
echo | openssl s_client -showcerts -connect raw.githubusercontent.com:443       
```                                                                             
That too works with /mnt/volumio and falls over with TOPDIR set to anything     else. Here it is falling over.
```                                                                             
CONNECTED(00000003)                                                             
depth=1 C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert SHA2     High Assurance Server CA
verify error:num=20:unable to get local issuer certificate                      
verify return:0                                                                 
---                                                                             
Certificate chain                                                               
 0 s:/C=US/ST=California/L=San Francisco/O=Fastly, Inc./CN=www.github.com       
   i:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 High Assurance   Server CA
-----BEGIN CERTIFICATE-----                                                     
...                                                                             
-----END CERTIFICATE-----                                                       
 1 s:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 High Assurance   Server CA
   i:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert High Assurance EV     Root CA
-----BEGIN CERTIFICATE-----                                                     
...                                                                             
-----END CERTIFICATE-----                                                       
```                                                                             
                                                                                
Here it is working                                                              
```                                                                             
CONNECTED(00000003)                                                             
depth=2 C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert High     Assurance EV Root CA
verify return:1                                                                 
depth=1 C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert SHA2     High Assurance Server CA
verify return:1                                                                 
depth=0 C = US, ST = California, L = San Francisco, O = "Fastly, Inc.", CN =    www.github.com
verify return:1                                                                 
---                                                                             
Certificate chain                                                               
 0 s:/C=US/ST=California/L=San Francisco/O=Fastly, Inc./CN=www.github.com       
   i:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 High Assurance   Server CA
-----BEGIN CERTIFICATE-----                                                     
...                                                                             
-----END CERTIFICATE-----                                                       
 1 s:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert SHA2 High Assurance   Server CA
   i:/C=US/O=DigiCert Inc/OU=www.digicert.com/CN=DigiCert High Assurance EV     Root CA
-----BEGIN CERTIFICATE-----                                                     
...                                                                             
-----END CERTIFICATE-----                                                       
```                                                                             
                                                                                
I made the build script stop just after the curl call and took a look around    for the two cases. Remember the only real code difference here is the location of the chroot. I found the /etc/ssl/certs directories have the same contents, in   particular the DigiCert High Assurance EV Root CA is there and its hash symlink is there with the correct value. There is no certificate for the 'DigiCert SHA2 High Assurance Server CA', but that's true in the working and non-working cases and in the host OS case.
                                                                                
I've run lots of grep -r looking for some place where /mnt/volumio is hardcoded that could have an impact on this but found nothing. It seems like a            misbehaviour of curl in this situation; I have to specify the location of the   certificate bundle file.
                                                                                
I found a debian bug (https://bugs.debian.org/814070) that looked similar.      
It was fixed for the reporter by updating to 7.46. One of the changes between 7.38 and 7.46 was this commit where they specify the full path to the certificate bundle -
https://anonscm.debian.org/cgit/collab-maint/curl.git/commit/?id=a494ae1901f86e03ed631f6aa6b0bf0758e75e35
which is essentially just what I had to do here. I dunno, it's a theory at least.

So to sum up, this PR can probably be closed but I'd like to submit one that adds the ```--cacert```  argument to the curl call, to clear the way for the TOPDIR thing. If you'd rather I didn't, just say so.
